storage_integration_test.go

```diff
@@ -69,41 +77,82 @@ func init() {
 func TestIntegration(t *testing.T) {
 	var err error
 
-	var k8sClient *kubernetes.Clientset
+	var restConfig *rest.Config
 	{
-		config := &rest.Config{
-			Host: server,
-			TLSClientConfig: rest.TLSClientConfig{
-				CertFile: crtFile,
-				KeyFile:  keyFile,
-				CAFile:   caFile,
-			},
+		c := k8srestconfig.DefaultConfig()
+
+		c.Logger = microloggertest.New()
+
+		c.Address = server
+		c.InCluster = false
+		c.TLS.CAFile = caFile
+		c.TLS.CrtFile = crtFile
+		c.TLS.KeyFile = keyFile
+
+		restConfig, err = k8srestconfig.New(c)
+		if err != nil {
+			panic(fmt.Sprintf("%#v", err))
 		}
+	}
+
+	g8sClient, err := versioned.NewForConfig(restConfig)
+	if err != nil {
+		panic(fmt.Sprintf("%#v", err))
+	}
 
-		k8sClient, err = kubernetes.NewForConfig(config)
+	k8sClient, err := kubernetes.NewForConfig(restConfig)
+	if err != nil {
+		panic(fmt.Sprintf("%#v", err))
+	}
+
+	k8sExtClient, err := apiextensionsclient.NewForConfig(restConfig)
+	if err != nil {
+		panic(fmt.Sprintf("%#v", err))
+	}
+
+	var crdClient *k8scrdclient.CRDClient
+	{
+		c := k8scrdclient.DefaultConfig()
+
+		c.K8sExtClient = k8sExtClient
+		c.Logger = microloggertest.New()
+
+		crdClient, err = k8scrdclient.New(c)
 		if err != nil {
-			t.Fatalf("error creating K8s client: %#v", err)
+			panic(fmt.Sprintf("%#v", err))
 		}
 	}
 
 	var storage *Storage
 	{
-		config := DefaultConfig()
-		config.K8sClient = k8sClient
-		config.Logger = microloggertest.New()
-
-		config.TPO.Name = "integration-test"
+		c := DefaultConfig()
+
+		c.CRDClient = crdClient
+		c.G8sClient = g8sClient
+		c.K8sClient = k8sClient
+		c.Logger = microloggertest.New()
+
+		c.Name = "integration-test"
+		c.Namespace = &v1.Namespace{
+			ObjectMeta: apismetav1.ObjectMeta{
+				Name:      "integration-test",
+				Namespace: "integration-test",
+			},
+		}
 
-		storage, err = New(config)
+		storage, err = New(c)
 		if err != nil {
 			t.Fatalf("error creating storage: %#v", err)
```

Comment:
You changed to panic everywhere but seemed to have missed this one.

Reply:
That was not intended. Fixed.